QBO Ce 3821

vinodborole opened the pull request:
## Highlights
* Quickbooks Online tests

tyler-cloud-elements commented:
@vinodborole Looks good! 👍 

Running the tests, I was getting some errors - please address:
31 passing (4m)
  12 pending
  8 failing

  1) bills should allow CRUDS for /hubs/finance/bills:

      AssertionError: expected status code 400 to equal 200 - {"requestId":"5858a8dfe4b052528ea4bd1b","message":"Bad Request","providerMessage":"2500 - Invalid Reference Id - Invalid Reference Id : Something you're trying to use has been made inactive. Check the fields with accounts, customers, items, vendors or employees."}
      + expected - actual

  2) bills should support searching /hubs/finance/bills by totalAmt:

      AssertionError: expected status code 400 to equal 200 - {"requestId":"5858a8e2e4b052528ea4bd1e","message":"Bad Request","providerMessage":"2500 - Invalid Reference Id - Invalid Reference Id : Something you're trying to use has been made inactive. Check the fields with accounts, customers, items, vendors or employees."}
      + expected - actual

  3) credit-memos should allow CRUDS for /hubs/finance/credit-memos:

      AssertionError: expected status code 400 to equal 200 - {"requestId":"5858a8e4e4b020f1b07a27d9","message":"Bad Request","providerMessage":"2500 - Invalid Reference Id - Invalid Reference Id : Line.SalesItemLineDetail.ItemRef"}
      + expected - actual

  4) credit-memos should support searching /hubs/finance/credit-memos by totalAmt:

      AssertionError: expected status code 400 to equal 200 - {"requestId":"5858a8e6e4b052528ea4bd23","message":"Bad Request","providerMessage":"2500 - Invalid Reference Id - Invalid Reference Id : Line.SalesItemLineDetail.ItemRef"}
      + expected - actual

  5) invoices should allow CRUDS for /hubs/finance/invoices:

      AssertionError: expected status code 400 to equal 200 - {"requestId":"5858a90ae4b052528ea4bd74","message":"Bad Request","providerMessage":"2500 - Invalid Reference Id - Invalid Reference Id : Line.SalesItemLineDetail.ItemRef2500 - Invalid Reference Id - Invalid Reference Id : Line.SalesItemLineDetail.ItemRef"}
      + expected - actual

  6) invoices should support searching /hubs/finance/invoices by totalAmt:

      AssertionError: expected status code 400 to equal 200 - {"requestId":"5858a90ce4b020f1b07a282f","message":"Bad Request","providerMessage":"2500 - Invalid Reference Id - Invalid Reference Id : Line.SalesItemLineDetail.ItemRef2500 - Invalid Reference Id - Invalid Reference Id : Line.SalesItemLineDetail.ItemRef"}
      + expected - actual

  7) payments should allow CRUDS for /hubs/finance/payments:

      AssertionError: expected status code 404 to equal 200 - {"requestId":"5858a918e4b052528ea4bd84","message":"Resource not found","providerMessage":"6000 - A business validation error has occurred while processing your request - Business Validation Error: Unexpected Internal Error. (-30035)"}
      + expected - actual


  8) sales-receipts should allow CRUDS for /hubs/finance/sales-receipts:

      AssertionError: expected status code 400 to equal 200 - {"requestId":"5858a92de4b052528ea4bd8b","message":"Bad Request","providerMessage":"2500 - Invalid Reference Id - Invalid Reference Id : Line.SalesItemLineDetail.ItemRef"}
      + expected - actual

tyler-cloud-elements commented:
@vinodborole Let me know if you help figuring these out

vinodborole commented:
@tyler-cloud-elements I ran the test again today; unfortunately i am not able to reproduce the errors you are seeing; here is the log of my tests.

debug: DELETE /hubs/finance/bills/199|0 with options %s
    ✓ should support searching /hubs/finance/bills by totalAmt (6897ms)

debug: DELETE /instances/31728 with options %s
debug: Deleted element instance with ID: 31728
debug: Modifying the chakram default request headers and base url

  3 passing (4m)




debug: DELETE /hubs/finance/credit-memos/193|0 with options %s
    ✓ should support searching /hubs/finance/credit-memos by totalAmt (8530ms)

debug: DELETE /instances/31724 with options %s
debug: Deleted element instance with ID: 31724
debug: Modifying the chakram default request headers and base url

  3 passing (10m)


debug: DELETE /hubs/finance/invoices/195|0 with options %s
    ✓ should support searching /hubs/finance/invoices by totalAmt (8543ms)

debug: DELETE /instances/31725 with options %s
debug: Deleted element instance with ID: 31725
debug: Modifying the chakram default request headers and base url

  3 passing (5m)



    ✓ should allow GET for /hubs/finance/sales-receipts (2265ms)

debug: DELETE /instances/31727 with options %s
debug: Deleted element instance with ID: 31727
debug: Modifying the chakram default request headers and base url

  2 passing (4m)

vinodborole commented:
@tyler-cloud-elements I am using my personal quickbook account for this test, can you share the one you are using along with the information of about the email id that is used to login to the quickbook account as quickbook requires OTP for sign-in. If possible please add my email Id "vinod.borole@gslab.com" to the alias.

tyler-cloud-elements commented:
@vinodborole ahh, that makes sense for all the reference errors. I was using the following for churros sauce. I'll look into trying to add you to the alias, but it should be coming to `	gslab@cloud-elements.com`. Do you have access to that?

"quickbooks": {
      "skip": true,
        "provisioning": "oauth1",
        "oauth.api.key": "qyprdxLrqNbUbzaIRo3AsvwE9bPwHO",
        "oauth.api.secret": "hfCIjrZK0dS7JiByOhCbq15UHvjsDn0Dq8hyy2zE",
        "username": "quickbooks@cloud-elements.com",
        "password": "Cl0udEl3m3ntsQB"
    },

vinodborole commented:
No i don't

tyler-cloud-elements commented:
@vinodborole Sorry for the late response on this. We've added you to the gslab@cloud-elements.com group, you should be getting emails for it now

ShitalPatil-GSLab commented:
@tyler-cloud-elements  Vinod is already part of  gslab@cloud-elements.com. Emails with OTP are not coming to account . Please add me to the group

tyler-cloud-elements commented:
Closing in favor of #530
